packages/flutter/lib/src/material/text_field.dart

```diff
@@ -567,6 +567,12 @@ class _TextFieldState extends State<TextField> with AutomaticKeepAliveClientMixi
     // TODO(jonahwilliams): uncomment out this check once we have migrated tests.
     // assert(debugCheckHasMaterialLocalizations(context));
     assert(debugCheckHasDirectionality(context));
+    final bool styleNotInherited = widget.style != null && widget.style.inherit == false;
```

Comment:
This might be a little clearer as one expression (we usually don't introduce variables that are only referred to in an assert). Also, the message should mention that it depends on `inherit false`.
```
assert(
  !(widget.style != null && widget.style.inherit == false &&
     (widget.style.fontSize == null || widget.style.textBaseline == null)),
  'inherit false style must supply fontSize and textBaseline',
);
```